rero_ils/query.py

```diff
@@ -163,6 +164,43 @@ def loans_search_factory(self, search, query_parser=None):
     return search, urlkwargs
 
 
+def templates_search_factory(self, search, query_parser=None):
+    """Template search factory.
+
+    Restricts results to organisation level for librarian and sys_lib.
+    Restricts results to private templates for users with role librarian.
+    """
+    search, urlkwargs = search_factory(self, search)
+    if current_patron:
+        if current_patron.is_system_librarian:
+            search = search.filter(
+                'term', organisation__pid=current_patron.get_organisation(
+                    )['pid'])
+        elif current_patron.is_librarian:
+            search = search.filter(
+                'term', organisation__pid=current_patron.get_organisation(
+                    )['pid'])
+            search = search.filter('bool', should=[
+                Q('bool', must=[
+                    Q(
+                        'match',
+                        creator__pid=current_patron.pid
```

Comment:
Too long for a single line ? It's a little bit difficult to read as an human :-)